go/pkg/metastore/coordinator/table_catalog.go

```diff
@@ -531,22 +531,26 @@ func (tc *Catalog) DeleteSegment(ctx context.Context, segmentID types.UniqueID)
 }
 
 func (tc *Catalog) UpdateSegment(ctx context.Context, updateSegment *model.UpdateSegment, ts types.Timestamp) (*model.Segment, error) {
+	if updateSegment.Collection == nil {
+		return nil, common.ErrMissingCollectionID
+	}
+
+	parsedCollectionID, err := types.Parse(*updateSegment.Collection)
```

Comment:
any chance of null pointer dereference? on updateSegment.Collection?

Reply:
What happens if i try to change collection?

Reply:
> any chance of null pointer dereference? on updateSegment.Collection?

no, we check if it's nil a few lines above

> What happens if i try to change collection?

by this do you mean changing the collection ID of a segment? this isn't currently supported, see comment here: https://github.com/chroma-core/chroma/pull/2613#discussion_r1700788277

Reply:
Yep, referring to the same thing. I think in that comment I propose that we should error or make the behavior explicit?